src/mumble/ServerHandler.cpp

```diff
@@ -392,6 +393,11 @@ void ServerHandler::sendPing() {
 	if (!connection)
 		return;
 
+	if (iInFlightTCPPings >= g.s.iMaxInFlightTCPPings) {
+		serverConnectionClosed(QAbstractSocket::UnknownSocketError, tr("Server is not responding to TCP pings"));
```

Comment:
I'd just like to add:

Calling `ServerHandler::disconnect()` sends a custom event, which is handled at https://github.com/mkrautz/mumble/blob/e5da2ba6daa7c77823c2a47adb6476c584992d0f/src/mumble/ServerHandler.cpp#L133

The disconnect method simply calls `exit(0)` to terminate the ServerHandler thread.

I'm happy to instead add a `reason` parameter to `disconnect` (or a `disconnectWithReason` method) -- as mentioned above.

That was the route I was taking until I found existing code that just called the serverConnectionClosed slot.